src/sidebar/components/test/thread-test.js

```diff
@@ -0,0 +1,284 @@
+import { mount } from 'enzyme';
+import { createElement } from 'preact';
+import { act } from 'preact/test-utils';
+
+import Thread from '../thread';
+import { $imports } from '../thread';
+
+import { checkAccessibility } from '../../../test-util/accessibility';
+import mockImportedComponents from '../../../test-util/mock-imported-components';
+
+// Utility functions to build nested threads
+let lastThreadId = 0;
+const createThread = () => {
+  lastThreadId++;
+  return {
+    id: lastThreadId.toString(),
+    annotation: {},
+    children: [],
+    parent: undefined,
+    collapsed: false,
+    visible: true,
+    depth: 0,
+    replyCount: 0,
+  };
+};
+
+const addChildThread = parent => {
+  const childThread = createThread();
+  childThread.parent = parent.id;
+  parent.children.push(childThread);
+  return childThread;
+};
+
+// NB: This logic lifted from `build-thread.js`
+function countRepliesAndDepth(thread, depth) {
```

Comment:
I realize this code was lifted from existing code in `build-thread`. It could be streamlined a little with modern syntax.

Reply:
You are right. fixing.